db/indexes.go

```diff
@@ -283,7 +282,7 @@ func (i *SGIndex) createIfNeeded(bucket base.N1QLStore, useXattrs bool, numRepli
 }
 
 // Initializes Sync Gateway indexes for bucket.  Creates required indexes if not found, then waits for index readiness.
-func InitializeIndexes(bucket base.N1QLStore, useXattrs bool, numReplicas uint, failFast bool) error {
```

Comment:
I would not have this take `bucket base.Bucket` parameter. This is about to change with CBG-2266,  but `base.Bucket` is an interface that is shared with `N1QLStore` (also an interface) so the really would be the same object and it's weird to pass them both in the same function. Also I think bucket is no longer needed now.

If I would have to pass both, I'd consider whether I could pass the interface that was the superset (base.Bucket) in this case.